src/main/java/pathfinding/ConcreteGraph.java

```diff
@@ -111,17 +111,24 @@ public int compareTo(ASTNode node) {
 	{
 		int i;
```

Comment:
This `i` variable is used only once, in a for loop far beneath this. Declare it in the for loop instead, that for loop currently just does `for (i = 0; [...])` when it could do `(int i = 0; [...])`